example-deployment/README.md

```diff
@@ -0,0 +1,329 @@
+
+For this example we will be using [rmf_demos](https://github.com/open-rmf/rmf_demos) as a local "deployment" of RMF, check that you have a working installation with
+
+```bash
+ros2 launch rmf_demos office.launch.xml
+```
+
+## kubernetes
+
+To start things off, we need an infrastructure to management our deployment. These days there are many public cloud services available, including new "serverless" architecture, but in this example, since we would like to be able to deploy in an airgapped network, all the public infrastructures are out.
+
+kubernetes has the advantage that it is very popular and there are several open source projets that allow you to quickly create your own cluster. In this example, we will be using `minikube` to create our own cluster.
+
+```bash
+./start-minikube.sh
+```
+
+This will download `minikube` to `.bin/minikube` and start a cluster.
+
+To stop the minikube cluster, run
+
+```bash
+.bin/minikube stop
+```
+
+## DNS
+
+In a real deployment, you would buy a domain and configure your dns provider to point it to your public ip. For this example, we will be deploying to `example.com`, but since we don't actually own the domain we will be using the `/etc/hosts` file to resolve it to our minikube ip.
+
+```bash
+if grep -q 'example.com' /etc/hosts; then
+  sudo sed "s/.*example.com.*/$(.bin/minikube ip) example.com/" /etc/hosts -i
+else
+  echo "$(.bin/minikube ip) example.com" | sudo tee -a /etc/hosts
+fi
+```
+
+This will create an entry in `/etc/hosts` to point `example.com` to the minikube cluster.
+
+Check that it works by pinging it
+
+```bash
+ping example.com
+```
+
+## SSL Certificate
+
+In a real deployment, you will get a verified cert, for this example, we will be using the default certs generated by minikube and keycloak. These are self-signed cert so you will get security warnings from the browser when you visit the urls, for the purpose of this example, you will need to proceed with the security exception.
+
+# Deploying rmf-web
+
+## Keycloak
+
+### Deploying
+
+In this example, we will be using [Keycloak](https://www.keycloak.org/) to do user management and authentication. It is an open source identity and access management, released under the Apache 2.0 license.
+
+There is a kubernetes resource file to deploy keycloak along with postgres for its database.
+
+```bash
+.bin/minikube kubectl -- apply -f k8s/keycloak.yaml
+```
+
+This requires internet connection, see [Deploying in an airgapped network](#deploying-in-an-airgapped-network) if you are in an airgap network. tldr:
+
+```bash
+# connected to internet
+docker pull quay.io/keycloak/keycloak:12.0.4
+
+# connected to airgap network
+.bin/minikube load quay.io/keycloak/keycloak:12.0.4
+.bin/minikube kubectl -- apply -f k8s/keycloak.yaml
+```
+
+Use kubectl to wait for keycloak to be ready
```

Comment:
maybe we could specify what to look out for? something like: look out for 0/1 to become 1/1 on the console output